src/Compilers/CSharp/Portable/Symbols/NullableAnnotationExtensions.cs

```diff
@@ -105,18 +124,25 @@ internal static ImmutableArray<ITypeSymbol> GetPublicSymbols(this ImmutableArray
         internal static ImmutableArray<CodeAnalysis.NullableAnnotation> ToPublicAnnotations(this ImmutableArray<TypeWithAnnotations> types) =>
             types.SelectAsArray(t => t.ToPublicAnnotation());
 
-        internal static CodeAnalysis.NullableAnnotation ToPublicAnnotation(TypeSymbol type, NullableAnnotation annotation) =>
-            annotation switch
+        internal static CodeAnalysis.NullableAnnotation ToPublicAnnotation(TypeSymbol type, NullableAnnotation annotation)
+        {
+            Debug.Assert(annotation != NullableAnnotation.Ignored);
+            return annotation switch
             {
-                CSharp.NullableAnnotation.Annotated => CodeAnalysis.NullableAnnotation.Annotated,
-                CSharp.NullableAnnotation.NotAnnotated => CodeAnalysis.NullableAnnotation.NotAnnotated,
+                NullableAnnotation.Annotated => CodeAnalysis.NullableAnnotation.Annotated,
+                NullableAnnotation.NotAnnotated => CodeAnalysis.NullableAnnotation.NotAnnotated,
+
                 // A value type may be oblivious or not annotated depending on whether the type reference
                 // is from source or metadata. (Binding using the #nullable context only when setting the annotation
                 // to avoid checking IsValueType early.) The annotation is normalized here in the public API.
-                CSharp.NullableAnnotation.Oblivious when type.IsValueType => CodeAnalysis.NullableAnnotation.NotAnnotated,
-                CSharp.NullableAnnotation.Oblivious => CodeAnalysis.NullableAnnotation.None,
+                NullableAnnotation.Oblivious when type.IsValueType => CodeAnalysis.NullableAnnotation.NotAnnotated,
+                NullableAnnotation.Oblivious => CodeAnalysis.NullableAnnotation.None,
+
+                NullableAnnotation.Ignored => CodeAnalysis.NullableAnnotation.None,
```

Comment:
what motivated the choice here to not throw UnexpectedValue in retail builds?

Reply:
> what motivated the choice here to not throw UnexpectedValue in retail builds?

Not worth crashing compiler over this

---
In reply to: [536442380](https://github.com/dotnet/roslyn/pull/49723#discussion_r536442380) [](ancestors = 536442380)

Reply:
Returning `None` would be the right behavior if we expected symbols to leak through public APIs in some way, so doesn't seem harmful to handle (I don't think it's worth crashing if we have such a bug). 
I don't feel very strongly about it

---
In reply to: [536442380](https://github.com/dotnet/roslyn/pull/49723#discussion_r536442380) [](ancestors = 536442380)